src/shared/fido2-util.c

```diff
@@ -0,0 +1,44 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "fido2-util.h"
+#include "fileio.h"
+#include "libfido2-util.h"
+#include "random-util.h"
+
+int fido2_generate_salt(struct iovec *ret_salt) {
+        _cleanup_(iovec_done) struct iovec salt = {};
```

Comment:
so, hmm, elsewhere you use iovec_done_erase for destroying the iovec of the salt.

The salt isn't precisely secret, is it? hence why bother with the erase stuff? I think it should be dropped everywhere, no?